src/SignalR/common/Shared/AsyncEnumerableAdapters.cs

```diff
@@ -24,13 +24,26 @@ public static IAsyncEnumerable<T> MakeCancelableTypedAsyncEnumerable<T>(IAsyncEn
             return new CancelableTypedAsyncEnumerable<T>(asyncEnumerable, cts);
         }
 
+#if NETCOREAPP3_0
```

Comment:
Is this not possible in `NETSTANDARD2_1`?

Reply:
`channel.ReadAllAsync` isn't in netstandard2.1 currently. And they seem to not be interested in moving corefx to netstandard2.1 currently. https://github.com/dotnet/corefx/issues/34963

Reply:
If you want an API added to netstandard2.1 you should request it on dotnet/standard, e.g. https://github.com/dotnet/standard/issues/1144

Reply:
Channels isn't in `netstandard` at all, it's a library that targets `netstandard2.0` (and doesn't cross-target 2.1).

My understanding of that issue though is that corefx isn't going to bulk-update to target 2.1. I think it doesn't preclude targeting 2.1 in a particular assembly.

If we really think `netstandard2.1` is worth it here then I guess this is why we need to triple-target. It just seems counter-productive to keep doing this.

Reply:
This is what the world of developing libraries that run everywhere is like.

btw I still think you should use feature flags. If makes this stuff easier https://github.com/aspnet/AspNetCore/pull/10540#discussion_r287878654

#VoiceOfExperience